### PR TITLE
[REVIEW] Apply 0.6.1 hotfix to current development branch; fixes CentOS issues

### PR DESCRIPTION
**Pull request by kkraus14**

### CODE REVIEW

File: cpp/src/hash/hashing.cu [hunk 1]
```diff
@@ -527,9 +527,8 @@ gdf_error hash_partition_gdf_table(gdf_table<size_type> const & input_table,
   cudf::table destination_table{partitioned_output.get_columns(),
                                 input_table.get_num_columns()};
 
-  auto gdf_status = cudf::detail::scatter(&source_table, row_output_locations,
-                                          &destination_table);
-  GDF_REQUIRE(GDF_SUCCESS == gdf_status, gdf_status);
+  cudf::detail::scatter(&source_table, row_output_locations,
+                        &destination_table);
```
Comment (thomcom) [inline cpp/src/hash/hashing.cu]: I hate to say it but shouldn't this be like a
```c++
#ifdef __CENTOS_BLAHBLAH
// base case
#else
// error checked case
#endif
```

Just to make sure we're writing real cross platform C++ here? 👅

Comment (kkraus14) [reply cpp/src/hash/hashing.cu]: @thomcom no, `cudf::detail::scatter` is a void function, this technically shouldn't even compile. It just so happens that Ubuntu returns a value aligned with success in our status enum whereas CentOS returns junk memory.

Comment (rlratzel) [reply cpp/src/hash/hashing.cu]: I think the problem is that `scatter()` returns void, meaning we wanted that change across the board anyway. Not sure why Ubuntu didn't have a problem with it, but (fortunately?) CentOS did and Keith was able to fix it.

Comment (thomcom) [reply cpp/src/hash/hashing.cu]: Oh, great! Thanks for setting me straight I didn't realize it is a `void`, I just assumed the original code made sense.

**Pull request merged**